virttest/utils_misc.py

```diff
@@ -3469,6 +3469,35 @@ def check_device_driver(pci_id, driver_type):
     return driver == driver_type
 
 
+def get_bootloader_cfg(session=None):
+    """
+    Find bootloader cfg file path of guest or host.
+
+    :param session: session of the vm needs to locate bootloader cfg file.
+    :return: bootloader cfg file path, empty string if no cfg file found.
+    """
+    bootloader_cfg = [
+        '/boot/grub/grub.conf',
+        '/boot/grub2/grub.cfg',
+        '/etc/grub.conf',
+        '/etc/grub2.cfg',
+        '/boot/etc/yaboot.conf'
+    ]
+    cfg_path = ''
+    for path in bootloader_cfg:
```

Comment:
how about update your code like below :)

test_func =  session and session.cmd_status or process.system
try:
     return [ cfg for  cfg in bootloader_cfg if not test_func("test -f %s" %  cfg)][-1]
except IndexError:
     logging.error("Failed to locate grub config file in %s." % bootloader_cfg)